File: api-gateway/src/main/java/kr/mafoo/gateway/WebClientConfig.java
package kr.mafoo.gateway;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.client.WebClient;

@Configuration
public class WebClientConfig {
    @Bean("externalWebClient")
    public WebClient externalServiceWebClient() {
        return WebClient.builder()
                .codecs(clientCodecConfigurer -> {
                    clientCodecConfigurer
                            .defaultCodecs()
                            .maxInMemorySize(16 * 1024 * 1024); // 16MB
                })
                .build();
    }
}

File: admin-service/src/main/kotlin/kr/mafoo/admin/config/WebClientConfig.kt
package kr.mafoo.admin.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.codec.ClientCodecConfigurer
import org.springframework.web.reactive.function.client.WebClient

@Configuration
class WebClientConfig(
    @Value("\${app.url.user-service}") private val userServiceUrl: String,
    @Value("\${app.url.photo-service}") private val photoServiceUrl: String,
) {
    @Bean("userServiceWebClient")
    fun userServiceWebClient(): WebClient {
        return WebClient
            .builder()
            .baseUrl(userServiceUrl)
            .codecs { clientCodecConfigurer: ClientCodecConfigurer ->
                clientCodecConfigurer
                    .defaultCodecs()
                    .maxInMemorySize(64 * 1024 * 1024) // 64MB
            }
            .build()
    }

    @Bean("photoServiceWebClient")
    fun photoServiceWebClient(): WebClient {
        return WebClient
            .builder()
            .baseUrl(photoServiceUrl)
            .codecs { clientCodecConfigurer: ClientCodecConfigurer ->
                clientCodecConfigurer
                    .defaultCodecs()
                    .maxInMemorySize(64 * 1024 * 1024) // 64MB
            }
            .build()
    }
}
